drivers/onewire/onewire.py: Fix undefined variable errors

amotl opened the pull request:
Dear Damien,

we are currently building a [test suite based on CPython/Pytest](https://github.com/hiveeyes/terkin-datalogger/tree/master/test) for invoking MicroPython programs. It works pretty good so far.

After spending some time adding tests for the telemetry domain, we would like to tackle the sensors domain. One thing we encountered there was that the file `drivers/onewire/onewire.py` actually raised `NameError: name 'SEARCH_ROM' is not defined` when being invoked on regular CPython3, see details below.

It looks like MicroPython is more forgiving here and while this looks like a cosmetic change / nit, we believe it is essentially about correctness and compatibility with CPython and should do no harm to MicroPython either.

So, we are humbly asking if you could consider merging this. This would enable us to continue our work while knowing that it might actually be also useful for others to be used with Genuine MicroPython.

With kind regards,
Andreas.

### Details
Invoking `pylint --errors-only` on this file yields:
```
E0602: Undefined variable 'MATCH_ROM' (undefined-variable)
E0602: Undefined variable 'SEARCH_ROM' (undefined-variable)
```

When invoking it on CPython, it croaks.
```
> self.writebyte(SEARCH_ROM)
E NameError: name 'SEARCH_ROM' is not defined
```

#### Editor highlighting
![image](https://user-images.githubusercontent.com/453543/77277679-c2245080-6cbd-11ea-9242-ef5919eaa807.png)

![image](https://user-images.githubusercontent.com/453543/77277689-c6e90480-6cbd-11ea-9814-7cdebef426ba.png)

amotl committed:
drivers/onewire/onewire.py: Fix undefined variable errors.

Invoking "pylint --errors-only" on this file yields::

    E0602: Undefined variable 'MATCH_ROM' (undefined-variable)
    E0602: Undefined variable 'SEARCH_ROM' (undefined-variable)

When invoking it on CPython::

    > self.writebyte(SEARCH_ROM)
    E NameError: name 'SEARCH_ROM' is not defined

While this looks like a cosmetic change / nit, it is essentially
about correctness and compatibility with CPython. It looks like
MicroPython is more forgiving here.

dpgeorge commented:
Thanks for the patch.  This code works in MicroPython because these variables are constants and the MicroPython parser/compiler optimises them out.  But indeed it is not valid Python because they are technically undefined within the scope they are used.

The fix here means that the constants will no longer be optimised.

The alternative fix would be to move these constant definitions to the global/module scope (instead of in the class scope).  That would mean they could still be optimised out.  But that changes the public API because these constants are intended to be accessible by users of the `OneWire` class, eg that is why `SKIP_ROM` is defined by never used.

So, it looks like the change in this PR is the right way forward.  But then all the const parts could be removed.

amotl committed:
drivers/onewire/onewire.py: Remove const definitions

After dca611a7, the "_ROM" constants stopped being 
optimised by the MicroPython parser/compiler anyway.
So, all const definitions can be removed completely.

amotl commented:
Dear Damien,

thanks for your guidance. So, I've removed all the `const`'s completely.

With kind regards,
Andreas.

---

P.S.: 
> After spending some time adding tests for the telemetry domain, we would like to tackle the sensors domain.

If you are interested, the outcome so far was this [1-Wire mock](https://github.com/hiveeyes/terkin-datalogger/blob/530ebfeb/test/util/micropython.py#L302-L340), which meets the specification of the corresponding [DS18B20 sensor test](https://github.com/hiveeyes/terkin-datalogger/blob/530ebfeb/test/test_sensors.py#L38-L39). Within the next iteration, this could be improved by rewiring its gist as a [real pytest fixture](https://github.com/hiveeyes/terkin-datalogger/tree/530ebfeb/test/fixtures). Then, the more generic fixture can easily be reused to mock out different kinds of 1-Wire devices efficiently.

dpgeorge commented:
Thanks for updating, squashed and merged in fbfea3b44007a2186468a57b0a27062d1060085d

dpgeorge commented:
> If you are interested, the outcome so far was this [1-Wire mock](https://github.com/hiveeyes/terkin-datalogger/blob/530ebfeb/test/util/micropython.py#L302-L340), which meets the specification of the corresponding [DS18B20 sensor test](https://github.com/hiveeyes/terkin-datalogger/blob/530ebfeb/test/test_sensors.py#L38-L39).

Thanks for pointing it out!
